01_fibonacci/fibi.go

```diff
@@ -0,0 +1,29 @@
+package main
+
+import (
+	"fmt"
+	"io"
+	"os"
+)
+
+var out io.Writer = os.Stdout
+
+//prints the fibonacci series
+func fibonacci(n int) {
+	a, b := 0, 1
+	if n > 0 {
+		fmt.Fprintln(out, a)
+	}
+	if n >= 1 {
+		fmt.Fprintln(out, b)
+	}
+	for i := 2; i <= n; i++ {
+		fmt.Fprintln(out, a+b)
+		b = a + b
+		a = b - a
```

Comment:
Clever! 🙂 However, you can avoid the need for this trickery by using multiple assignment: `a, b = …`.

Reply:
updated

Reply:
Achievement unlocked: received @anzdaddy `clever` label
(I've been working on this for over a year now to no avail).